lib/api/src/grpc/proto/points.proto

```diff
@@ -354,32 +369,47 @@ message CountPoints {
 message PointsUpdateOperation {
   message PointStructList {
     repeated PointStruct points = 1;
+    optional ShardKeySelector shard_key_selector = 2; // Option for custom sharding to specify used shard keys
   }
   message SetPayload {
       map<string, Value> payload = 1;
       optional PointsSelector points_selector = 2; // Affected points
+      optional ShardKeySelector shard_key_selector = 3; // Option for custom sharding to specify used shard keys
   }
   message DeletePayload {
       repeated string keys = 1;
       optional PointsSelector points_selector = 2; // Affected points
+      optional ShardKeySelector shard_key_selector = 3; // Option for custom sharding to specify used shard keys
   }
   message UpdateVectors {
     repeated PointVectors points = 1; // List of points and vectors to update
+    optional ShardKeySelector shard_key_selector = 2; // Option for custom sharding to specify used shard keys
   }
   message DeleteVectors {
     PointsSelector points_selector = 1; // Affected points
     VectorsSelector vectors = 2; // List of vector names to delete
+    optional ShardKeySelector shard_key_selector = 3; // Option for custom sharding to specify used shard keys
+  }
+  message DeletePoints {
+    PointsSelector points = 1; // Affected points
+    optional ShardKeySelector shard_key_selector = 2; // Option for custom sharding to specify used shard keys
+  }
+  message ClearPayload {
+    PointsSelector points = 1; // Affected points
+    optional ShardKeySelector shard_key_selector = 2; // Option for custom sharding to specify used shard keys
   }
 
   oneof operation {
     PointStructList upsert = 1;
-    PointsSelector delete = 2;
+    PointsSelector delete_deprecated = 2 [deprecated=true];
     SetPayload set_payload = 3;
     SetPayload overwrite_payload = 4;
     DeletePayload delete_payload = 5;
-    PointsSelector clear_payload = 6;
+    PointsSelector clear_payload_deprecated = 6 [deprecated=true];
```

Comment:
Have to deprecate a couple of fields here and replace it with more consistent version of data structures.
Backward compatibility should still work, but it would require migration in python client